google-api-client/src/main/java/com/google/api/client/googleapis/util/Utils.java

```diff
@@ -55,6 +71,82 @@ public static HttpTransport getDefaultTransport() {
     static final HttpTransport INSTANCE = new NetHttpTransport();
   }
 
+  /**
+   * Returns the `cert_provider_command` field in context_aware_metadata.json file.
+   * 
+   * @param contextAwareMetadataJson ~/.secureConnect/context_aware_metadata.json file content.
+   * @return `cert_provider_command` field
+   * @since 1.31
+   */
+  @SuppressWarnings("unchecked")
+  public static ArrayList<String> extractCertificateProviderCommand(String contextAwareMetadataJson) {
+    GsonBuilder builder = new GsonBuilder();
+    LinkedTreeMap<String, Object> map = (LinkedTreeMap<String, Object>)builder.create().fromJson(contextAwareMetadataJson, Object.class);
+    return (ArrayList<String>)map.get("cert_provider_command");
+  }
+
+  /**
+   * Returns the default client certificate by running the cert_provider_command commands
+   * from ~/.secureConnect/context_aware_metadata.json file.
+   * 
+   * @return The default client certificate input stream
+   * @since 1.31
+   */
+  public static InputStream loadDefaultCertificate() throws IOException, GeneralSecurityException {
+    File file = new File(CONTEXT_AWARE_METADATA_PATH);
+    if (!file.exists()) {
+      return null;
+    }
+
+    // Load the cert provider command from the json file.
+    String json = new String(Files.readAllBytes(Paths.get(CONTEXT_AWARE_METADATA_PATH)));
+    ArrayList<String> command = extractCertificateProviderCommand(json);
+    
+    // Call the command.
+    Process process = new ProcessBuilder(command).start();
+    int exitCode = 0;
+    try {
+      exitCode = process.waitFor();
+    } catch (InterruptedException exception) {
+      throw new GeneralSecurityException("Failed to execute cert provider command", exception);
```

Comment:
This is probably not the right exception. `GeneralSecurityException` is for generally for crypto related security issues. Perhaps (`AccessControlException`)[https://docs.oracle.com/javase/7/docs/api/java/security/AccessControlException.html]?

Reply:
changed to `AccessControlException`.